tests/rptest/tests/delete_retention_ms_test.py

```diff
@@ -0,0 +1,222 @@
+from rptest.util import expect_exception
+
+
+class DeleteRetentionMsTest(RedpandaTest):
+    def __init__(self, ctx):
+        self.ctx = ctx
+        super().__init__(ctx)
+
+        self.rpk = RpkTool(self.redpanda)
+
+    @cluster(num_nodes=1)
+    def test_get_cluster_config(self):
+        # Default value
+        cluster_default = self.rpk.cluster_config_get('tombstone_retention_ms')
+        assert cluster_default == 'null'
+
+    @cluster(num_nodes=1)
+    def test_set_cluster_config(self):
+        self.rpk.cluster_config_set('tombstone_retention_ms', 1234567890)
+        cluster_prop = self.rpk.cluster_config_get('tombstone_retention_ms')
+        assert cluster_prop == "1234567890"
+
+        topic_name = "tapioca"
+        topic = TopicSpec(name=topic_name)
+        self.rpk.create_topic(topic_name, partitions=1)
+        topic_desc = self.rpk.describe_topic_configs(topic_name)
+        assert topic_desc['delete.retention.ms'][0] == '1234567890'
+
+    @cluster(num_nodes=1)
+    def test_alter_topic_config(self):
+        topic_name = "tapioca"
+        topic = TopicSpec(name=topic_name)
+        self.rpk.create_topic(topic_name, partitions=1)
+        topic_desc = self.rpk.describe_topic_configs(topic_name)
+
+        # Upon topic construction, this property is actually "disabled" by default.
+        assert topic_desc['delete.retention.ms'][0] == '-1'
+        assert topic_desc['delete.retention.ms'][1] == 'DEFAULT_CONFIG'
+
+        # Set value
+        self.rpk.alter_topic_config(topic_name, 'delete.retention.ms',
+                                    1234567890)
+
+        topic_desc = self.rpk.describe_topic_configs(topic_name)
+        assert topic_desc['delete.retention.ms'][0] == '1234567890'
+        assert topic_desc['delete.retention.ms'][1] == 'DYNAMIC_TOPIC_CONFIG'
+
+        # Set cluster value and delete topic config
+        self.rpk.cluster_config_set('tombstone_retention_ms', 100)
+        self.rpk.delete_topic_config(topic_name, 'delete.retention.ms')
+
+        topic_desc = self.rpk.describe_topic_configs(topic_name)
+        assert topic_desc['delete.retention.ms'][0] == '100'
+        assert topic_desc['delete.retention.ms'][1] == 'DEFAULT_CONFIG'
+
+        # Disable topic value
+        self.rpk.alter_topic_config(topic_name, 'delete.retention.ms', -1)
+        topic_desc = self.rpk.describe_topic_configs(topic_name)
+        assert topic_desc['delete.retention.ms'][0] == '-1'
+        assert topic_desc['delete.retention.ms'][1] == 'DYNAMIC_TOPIC_CONFIG'
+
+    @cluster(num_nodes=1)
+    def test_create_topic_configurations_cluster_defaults(self):
+        topic_name = "tapioca"
+        TopicAndClusterConf = namedtuple('TopicAndClusterConf', [
+            'delete_retention_ms', 'cloud_storage_remote_read',
+            'cloud_storage_remote_write', 'valid'
+        ])
+        #We cannot enable delete.retention.ms if any cloud storage properties are also enabled.
+        test_cases = [
```

Comment:
Does this cover the entirety of the truth table from the earlier commit? Might be worth duplicating the truth table here

Reply:
Also nit: space between # and the first word

Reply:
No it does not, and the test was not really intended to test the truth table for the overrides- more so tiered storage and `delete.retention.ms` property interaction.

Reply:
Got it. Does it make sense to include a test case for just the `delete.retention.ms` behavior without tiered storage involvement?

Reply:
[test_alter_topic_config](https://github.com/redpanda-data/redpanda/blob/877b01f8f4154ce36c8738fde5e606d2f2e1aed0/tests/rptest/tests/delete_retention_ms_test.py#L45) covers this already I think.